src/Control/Eff/ImplementYourOwnEffect.hs

```diff
@@ -0,0 +1,33 @@
+-- | This module exports functions, types, and typeclasses necessary for
+-- implementing a custom effect and/or effect handler
+module Control.Eff.ImplementYourOwnEffect
```

Comment:
@power-fungus how do you feel about `Control.Eff.Extend` as the module name?

Reply:
@power-fungus agreed. But I feel good about it. It might also help to internalize internal effect implementations to explicitly use `Control.Eff.Extend`. What do you think?

Reply:
@suhailshergill do you suggest implementing all effects of this library via the use of the `Extend` module? I would be in favour of that.

Reply:
precisely :)
:+1:

Reply:
@power-fungus and to be clear all functions/definitions today in `Internal` would get exported from one of the following ways:
- via `Control.Eff` => some subset of exports which are needed for using currently defined effects
- via `Control.Eff.Lift` => `(Lift(..), lift, runLift)`
- - via `Control.Eff.Extend` => everything else (`interpose`, `raise`, ...)

Reply:
@suhailshergill It is the intent that all functions are exported one way or another this is especially enforced if the effect-modules do not import `Internal` directly.